src/util/CompositeAsyncHandler.ts

```diff
@@ -0,0 +1,42 @@
+import { AsyncHandler } from './AsyncHandler';
+import { UnsupportedHttpError } from './errors/UnsupportedHttpError';
+
+export class CompositeAsyncHandler<TIn, TOut> implements AsyncHandler<TIn, TOut> {
+  private readonly handlers: AsyncHandler<TIn, TOut>[];
+
+  public constructor (handlers: AsyncHandler<TIn, TOut>[]) {
+    this.handlers = handlers;
+  }
+
+  public async canHandle (input: TIn): Promise<void> {
+    const errors: Error[] = [];
+
+    for (const handler of this.handlers) {
+      try {
+        await handler.canHandle(input);
+
+        return;
+      } catch (error) {
+        errors.push(error);
+      }
+    }
+
+    throw new UnsupportedHttpError(errors.map((error: Error): string => error.message).join(' '));
```

Comment:
I would join with a newline, and prepend with a more generic error message. Something like "All composed handlers failed:\n".

Reply:
The main reason I didn't use newlines was just in case further down the line something else wanted to add to the error message (like another composite error for example). So perhaps still keep them on one line? E.g. "All composed handlers failed: [${join stuff}]".

Reply:
Ah yes, that's true. The `[]` wrapping sounds good, but perhaps join with `, ` then?